### PR TITLE
Claim not found

### PR DESCRIPTION
**Pull request by kaapstorm**
- Better handling of claiming a case that doesn't exist
- Fixes test
- Changes "claim" case type to "commcare-case-claim" to make it less likely that a domain might use the same case type, and avoid possible confusion.

@proteusvacuum cc @czue

### CODE REVIEW

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 1]
```diff
@@ -1,37 +1,54 @@
...
 
 DOMAIN = 'test-domain'
 USERNAME = 'testy_mctestface'
 PASSWORD = '123'
 CASE_NAME = 'Jamie Hand'
 CASE_TYPE = 'case'
+OWNER_ID = 'nerc'
+TIMESTAMP = '2016-04-17T10:13:06.588694Z'
+PATTERN = r'\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d+Z'
+# because http://www.theguardian.com/environment/2016/apr/17/boaty-mcboatface-wins-poll-to-name-polar-research-vessel
```
Comment (dimagimon) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: E501 line too long (117 > 115 characters)

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 2]
```diff
@@ -69,21 +86,28 @@ def test_duplicate_claim(self):
...
-        <type>{case_type}</type>
-    </case>
-</fixture>""".format(
-            case_id=self.case_id,
-            case_name=CASE_NAME,
-            case_type=CASE_TYPE
-        )
+        known_result = (
+            '<fixture id="case">'  # (this is not the case type)
+                '<case case_id="{case_id}" '
```
Comment (dimagimon) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: E131 continuation line unaligned for hanging indent

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 3]
```diff
@@ -69,21 +86,28 @@ def test_duplicate_claim(self):
...
+                      'case_type="{case_type}" '
```
Comment (dimagimon) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: E131 continuation line unaligned for hanging indent

File: corehq/apps/ota/tests/test_search_claim_endpoints.py [hunk 4]
```diff
@@ -69,21 +86,28 @@ def test_duplicate_claim(self):
...
+                      'owner_id="{owner_id}" '
+                      'status="open">'
+                    '<case_name>{case_name}</case_name>'
```
Comment (dimagimon) [inline corehq/apps/ota/tests/test_search_claim_endpoints.py]: E131 continuation line unaligned for hanging indent

**Comment by proteusvacuum** [comment]
:+1: on :white_check_mark:

**Comment by proteusvacuum** [comment]
:boat:

**Pull request merged**